low-level/actuators/impl/centos_7/systemd_service.py

```diff
@@ -126,42 +123,78 @@ def perform_request(self, jsonMsg):
 
         # Give the unit some time to finish starting/stopping to get final status
         time.sleep(5)
 
-        # Get the current status of the process and return it back if no result
-        if result is None:
-            state = self._get_activestate()
-            substate = self._get_active_substate()
-            self._log_debug("perform_request, state: %s, substate: %s" %
+        # Get the current status of the process and return it back:
+        state, substate, pid, command_line = Service('dbus').get_service_information(self._service_name)
+        status = Service('dbus').check_service_is_enabled(self._service_name)
+        # Parse dbus output to fetch command line path with args.
+        command_line_path_with_args = []
+        for field in list(command_line[0][1]):
+            command_line_path_with_args.append(str(field))
+        timestamp = get_service_timestamp(self._service_name)
+        result["state"] = state
+        result["substate"] = substate
+        result["status"] = status
+        result["pid"] = pid
+        result["timestamp"] = timestamp
+        result["command_line_path"] = command_line_path_with_args
+
+        logger.debug("perform_request, state: %s, substate: %s" %
                                 (str(state), str(substate)))
-            return (self._service_name, state, substate)
-
-        return (self._service_name, str(result), None)
-
-    def _get_activestate(self):
-        """"Returns the active state of the unit"""
-        return self._proxy.Get('org.freedesktop.systemd1.Unit',
-                                'ActiveState',
-                                dbus_interface='org.freedesktop.DBus.Properties')
-
-    def _get_active_substate(self):
-        """"Returns the active state of the unit"""
-        return self._proxy.Get('org.freedesktop.systemd1.Unit',
-                                'SubState',
-                                dbus_interface='org.freedesktop.DBus.Properties')
+        return (self._service_name, result, is_err_response)
+
+def parse_enable_disable_dbus_result(dbus_result):
+    """Parse debus output and add in dictionary."""
+    # Parse dbus output.
+    # dbus o/p: dbus.Array([dbus.Struct((dbus.String('symlink'),
+    # dbus.String('/etc/systemd/system/multi-user.target.wants/statsd.service'),
+    # dbus.String('/usr/lib/systemd/system/statsd.service')), signature=None)], signature=dbus.Signature('(sss)'))
+    result = {}
+    key = None
+    if list(dbus_result):
+        dbus_result = list(dbus_result[0])
+        for field in dbus_result:
+            if field in ["symlink", "unlink"]:
+                key = field
+                result[key] = []
+                continue
+            elif field:
+                result[key].append(str(field))
+    return result
+
+def get_service_timestamp(service_name):
+    """Get service timestamp."""
+    timestamp = None
+    cmd = f"systemctl status {service_name} "

```

Comment:
This function serves to get whole status only, which is replica of function in cortx-utils using dbus api? Can convert response from that dbus api to present in a way we need, like systemctl ?

Reply:
Checked it but I couldn't get way to convert dbus timestamp response to day-date-time format- like systemctl response.